template.go

```diff
@@ -94,6 +94,16 @@ func AddFuncMap(key string, fn interface{}) error {
 	return nil
 }
 
+// BeegoTplFuncMap returns a copy of the beego template func map, which may then be used for
+// rendering templates directly (e.g., with another template engine).
+func BeegoTplFuncMap() template.FuncMap {
+	mapCopy := make(template.FuncMap)
```

Comment:
1.not thread safe
2.make with capacity
3.beego support template engine https://github.com/astaxie/beego/pull/1762